frontend/src/metabase/common/hooks/use-entity-list-query/use-entity-list-query.unit.spec.tsx

```diff
@@ -169,4 +177,41 @@ describe("useEntityListQuery", () => {
 
     expect(screen.getByText(TEST_TABLE.name)).toBeInTheDocument();
   });
+
+  it("should not remove loader in case second api call is cached", async () => {
+    const dbsResponse = [TEST_DB];
+    fetchMock.get(
+      "path:/api/database",
+      delay(100).then(() => dbsResponse),
+      { overwriteRoutes: true },
+    );
+
+    const { rerender } = setup();
+
+    expect(fetchMock.calls("path:/api/database")).toHaveLength(1);
+
+    rerender(
+      <>
+        <TestComponent />
+        <TestComponent testId="test2" />
+      </>,
+    );
+
+    // second component should not create extra request, make sure that caching works as expected
+    expect(fetchMock.calls("path:/api/database")).toHaveLength(1);
+
+    expect(
+      within(screen.getByTestId("test2")).getByTestId("loading-spinner"),
+    ).toBeInTheDocument();
+
+    await delay(100);
+
+    expect(fetchMock.calls("path:/api/database")).toHaveLength(1);
+    expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
+
+    await delay(0); // trigger delay to make sure state has been updated
+
+    expect(fetchMock.calls("path:/api/database")).toHaveLength(1);
+    expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
```

Comment:
Is this needed? Because we already `await delay(100)` above, which should be enough for request(s) to start/finish.